src/details/ArborX_DetailsFDBSCANDenseBox.hpp

```diff
@@ -238,7 +188,7 @@ struct FDBSCANDenseBoxCallback
   }
 };
 
-template <typename ExecutionSpace, typename Primitives>
+template <typename ExecutionSpace, typename Primitives, typename CartesianGrid>
```

Comment:
Why did you template that guy?

Reply:
It takes `CartesianGrid` as an argument.

Reply:
My point is you can deduce the type entirely.
Primitives -> DIM -> CartesianGrid

Reply:
That's fair.